### ISSUE TITLE
Admin Account Creation

### ISSUE THREAD

**Issue by WhiteTiburon** [RocketChat/Rocket.Chat#29855]
### Description:

I am attempting to create an Initial user via the compose file following **https://docs.rocket.chat/setup-and-configure/accessing-your-workspace/creating-the-first-administrator** and I get a 

=== UnHandledPromiseRejection ===
errorClass [Error]: Match error: Expected string, got object
    at check (packages/check/match.js:36:17)
    at packages/accounts-password/password_server.js:324:3

 It leads back to the password not being sent as a string. However, I made sure to make the password a string and still get the same error.

### Steps to reproduce:

1. add variables to compose file
2. run compose file via docker-compose up -d

### Expected behavior:

Creates user and I am able to log in.

### Actual behavior:
I get this error in the logs:

Inserting admin user:
Name: Administrator
Email: me@me.com
Username: Jon
=== UnHandledPromiseRejection ===
errorClass [Error]: Match error: Expected string, got object
    at check (packages/check/match.js:36:17)
    at packages/accounts-password/password_server.js:324:3
    at /app/bundle/programs/server/npm/node_modules/meteor/promise/node_modules/meteor-promise/fiber_pool.js:43:40
 => awaited here:
    at Function.Promise.await (/app/bundle/programs/server/npm/node_modules/meteor/promise/node_modules/meteor-promise/promise_server.js:56:12)
    at server/startup/initialData.js:122:4
    at /app/bundle/programs/server/npm/node_modules/meteor/promise/node_modules/meteor-promise/fiber_pool.js:43:40 {
  path: '',
  sanitizedError: errorClass [Error]: Match failed [400]
      at errorClass.<anonymous> (packages/check/match.js:91:27)
      at new errorClass (packages/meteor.js:692:17)
      at check (packages/check/match.js:36:17)
      at packages/accounts-password/password_server.js:324:3
      at /app/bundle/programs/server/npm/node_modules/meteor/promise/node_modules/meteor-promise/fiber_pool.js:43:40 {
    isClientSafe: true,
    error: 400,
    reason: 'Match failed',
    details: undefined,
    errorType: 'Meteor.Error'
  },
  errorType: 'Match.Error'
}

Here is what I added to the compose file:

ADMIN_USERNAME: test
ADMIN_PASS: "password"
ADMIN_EMAIL: me@me.com

![Screenshot 2023-07-18 at 11 00 17 AM](https://github.com/RocketChat/Rocket.Chat/assets/46905891/1fc3246a-0cca-4df6-9d21-1f5992ef5a2b)
![Screenshot 2023-07-18 at 11 00 59 AM](https://github.com/RocketChat/Rocket.Chat/assets/46905891/5282f499-bc69-4a64-907f-ddfa9dc88eab)

### Server Setup Information:
                                        
|  Rocket.Chat Version: 6.2.0                                       
|       NodeJS Version: 14.21.2 - x64                               
|      MongoDB Version: 5.0.18                                      
|       MongoDB Engine: wiredTiger                                  
|             Platform: linux                                       
|         Process Port: 5182                                        
|             Site URL: http://localhost:5182  
|     ReplicaSet OpLog: Enabled                                     
|          Commit Hash: d918834d31                                  
|        Commit Branch: HEAD

**Comment by hugocostadev** [comment]
Did you tried to create the admin thought the Setup Wizard step? It's easier and the most common flow that users do

---
**Questions? Help needed? Feature Requests?** 

- Join our [Open Server](https://open.rocket.chat/) in the #support channel and feel free to raise a question
- Join our [Community Forum](https://forums.rocket.chat/) and search/create a post there
- Feature Request: Open an issue in [Feature Request](https://github.com/RocketChat/feature-requests) repository

**Comment by Dnouv** [comment]
Hey @WhiteTiburon 

I am unsure of the exact format you used for the compose file. But could you please try this (notice the no quotes around the password):
```compose
environment:
      - ADMIN_USERNAME=test
      - ADMIN_PASS=password
      - ADMIN_EMAIL=me@me.com
```

Thank you!

**Comment by WhiteTiburon** [comment]
> 

Yea, but want to avoid that step.

**Comment by debdutdeb** [comment]
we have a bug here

**Comment by debdutdeb** [comment]
![image](https://github.com/RocketChat/Rocket.Chat/assets/76006232/888195d4-26cc-4157-a795-24ee30b84595)

```
setPasswordAsync(id.insertedId, ['admin'])
```

**Comment by WhiteTiburon** [comment]
> ![image](https://user-images.githubusercontent.com/76006232/255989093-888195d4-26cc-4157-a795-24ee30b84595.png)
> 
> ```
> setPasswordAsync(id.insertedId, ['admin'])
> // file: initialData.js
> ```

What should I do?

**Comment by engelgabriel** [comment]
Looks like a simple bug to fix. @debdutdeb  can you provide a pull request?

**Comment by engelgabriel** [comment]
> Did you tried to create the admin thought the Setup Wizard step? It's easier and the most common flow that users do

This suggestions does not solve the problem. The admin creating automation via environment variable needs to work. We must fix the bug.

**Comment by lamberms** [comment]
I am having the same issues with the newest docker container.  For my use case this is necessary as my system comes up fresh every run with Docker volumes wiped.  We can (with pain) get around it for now by setting up the first user every time we run.

**Comment by WhiteTiburon** [comment]
Since it is being merged to develop does this mean the fix will not be included in 6.2.10?
